src/proxy/ddl.go

```diff
@@ -36,17 +36,35 @@ func CheckCreateTable(ddl *sqlparser.DDL) error {
 		return fmt.Errorf("spanner.ddl.check.create.table[%s].error:not support", table)
 	}
 
-	// UNIQUE/PRIMARY constraint check.
+	// shard key and UNIQUE/PRIMARY KEY constraint check.
 	if shardKey != "" {
 		shardKeyOK := false
+		// constraint check in column definition
 		for _, col := range ddl.TableSpec.Columns {
 			colName := col.Name.String()
 			if colName == shardKey {
 				shardKeyOK = true
 			} else {
 				switch col.Type.KeyOpt {
 				case sqlparser.ColKeyUnique, sqlparser.ColKeyUniqueKey, sqlparser.ColKeyPrimary:
-					return fmt.Errorf("The unique/primary constraint only be defined on the sharding key column[%s] not [%s]", shardKey, colName)
+					return fmt.Errorf("A UNIQUE INDEX must include all columns in the table's partitioning function")
```

Comment:
The error messsage is not good for users, because we not only handle UNIQUE, but also PRMARY or others.
Please make sure what's you are doing before we do this.

Reply:
The shardKey, colName need print to debug

Reply:
The shardKey had been printed out, but I don`t think we need to print out the constraint columns that diff with shardKey, for such case:
```
create table tbl( a int, b char, c int, d datetime, primary key(b, c, d)) partition by hash(a);
```
or

```
create table tbl( a int , b char unique, c int primary key, d datetime primary key) partition by hash(a);
```